{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Purpose:** Extract the details about the current resources in the Galaxy Training Material and some statistics"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'2020-09-28 15:16:48.123679'"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import base64\n",
    "import copy\n",
    "import datetime\n",
    "import matplotlib.pyplot as plt\n",
    "import pandas as pd\n",
    "import re\n",
    "import yaml\n",
    "\n",
    "from dateutil import relativedelta\n",
    "\n",
    "from github import Github\n",
    "from pprint import pprint\n",
    "\n",
    "\n",
    "str(datetime.datetime.now())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Usage:\n",
    "1. Generate a Personal access tokens on GitHub (in Setting)\n",
    "2. Add it to the `../config.yaml` after `github:`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(\"../config.yaml\", \"r\") as stream:\n",
    "    config = yaml.load(stream, Loader=yaml.FullLoader)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# connect to GitHub\n",
    "g = Github(config[\"github\"])\n",
    "# retrieve the hub repository\n",
    "repo = g.get_user(\"galaxyproject\").get_repo(\"training-material\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# dates and corresponding commit\n",
    "gitcommit_dates = {\n",
    "    \"22.06.17\": \"44164c9b\",\n",
    "    \"22.06.18\": \"24ea24ef\",\n",
    "    \"22.06.19\": \"4ae23d58\",\n",
    "    \"22.06.20\": \"246db10d\",\n",
    "    'today': \"master\"}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Evolution of topics, tutorials and technical support"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "# create range per month\n",
    "start = datetime.datetime(year=2017, month=6, day=1)\n",
    "end = datetime.datetime.now()\n",
    "r = relativedelta.relativedelta(end, start)\n",
    "month_nb = 12*r.years + r.months\n",
    "months = [start + relativedelta.relativedelta(months=i) for i in range(month_nb+1)]\n",
    "months.append(end)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [],
   "source": [
    "# get commit sha of latest commit before beginning of each month\n",
    "commits = {}\n",
    "for m in months:\n",
    "    commits[m] = repo.get_commits(since=m-relativedelta.relativedelta(months=1),until=m)[0].sha"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [],
   "source": [
    "def has_zenodo(file_content):\n",
    "    '''\n",
    "    Check if Zenodo link in tutorial\n",
    "\n",
    "    :param file_content: the encoded content of the file obtained with GitHub API\n",
    "\n",
    "    :return: boolean\n",
    "    '''\n",
    "    decoded_file_content = base64.b64decode(file_content)\n",
    "    return 'zenodo_link' in str(decoded_file_content)\n",
    "\n",
    "\n",
    "def is_enable(file_content):\n",
    "    '''\n",
    "    Check if tutorial is enabled or not link in tutorial\n",
    "\n",
    "    :param file_content: the encoded content of the file obtained with GitHub API\n",
    "\n",
    "    :return: boolean\n",
    "    '''\n",
    "    decoded_file_content = base64.b64decode(file_content)\n",
    "    return not 'enable: ' in str(decoded_file_content)\n",
    "\n",
    "\n",
    "def get_topic_title(topic_n, ref):\n",
    "    '''\n",
    "    :param topic_n: name of the topic\n",
    "    :param ref: name of the commit/branch/tag\n",
    "    \n",
    "    :return: title of the topic in metadata file\n",
    "    '''\n",
    "    metadata_f = repo.get_contents(\"topics/%s/metadata.yaml\" % topic_n, ref)\n",
    "    metadata_f_content = yaml.load(base64.b64decode(metadata_f.content), Loader=yaml.FullLoader)\n",
    "    return metadata_f_content['title']\n",
    "    \n",
    "\n",
    "def extract_training_content(ref):\n",
    "    '''\n",
    "    Extract the details about the current resources in the Galaxy Training Material,\n",
    "    and aggregate information per topics\n",
    "\n",
    "    :param ref: name of the commit/branch/tag\n",
    "\n",
    "    :return: dictionary with information about the tutorials (tutorial, slides, workflow, \n",
    "    data library) in each topic on ref and aggregated statistics per topics\n",
    "    '''\n",
    "    trainings = {}\n",
    "    topics_dir = \"topics\"\n",
    "    infos = {'workflows': False, 'zenodo': False}\n",
    "    stat_c = {'introduction': 0, 'tutorials': 0, 'workflows': 0, 'zenodo': 0}\n",
    "    stats = pd.DataFrame(columns=list(stat_c.keys()), index=[])\n",
    "    for topic in repo.get_contents(\"topics\", ref):\n",
    "        # get topic details\n",
    "        topic_n = topic.name\n",
    "        print(topic_n)\n",
    "        topic_t = get_topic_title(topic_n, ref)\n",
    "        # prepare aggregation for topic\n",
    "        trainings.setdefault(topic_n, {})\n",
    "        topic_stat = copy.copy(stat_c)\n",
    "        # check for introduction\n",
    "        has_tuto_folder = False\n",
    "        for topic_c in repo.get_contents(\"%s/\" % topic.path, ref):\n",
    "            if topic_c.name == 'slides':\n",
    "                topic_stat['introduction'] += len(\n",
    "                    repo.get_contents(\"%s/slides/\" % topic.path, ref))\n",
    "            if topic_c.name == 'tutorials':\n",
    "                has_tuto_folder = True\n",
    "        if not has_tuto_folder:\n",
    "            continue\n",
    "        # parse the tutorials\n",
    "        for tuto in repo.get_contents(\"%s/tutorials/\" % topic.path, ref):\n",
    "            tuto_n = tuto.name\n",
    "            if '.md' in tuto_n:\n",
    "                continue\n",
    "            print(tuto_n)\n",
    "            enabled = True\n",
    "            trainings[topic_n].setdefault(tuto_n, copy.copy(infos))\n",
    "            # parse the files in the tuto folder\n",
    "            for tuto_f in repo.get_contents(tuto.path, ref):\n",
    "                if tuto_f.name == 'tutorial.md':\n",
    "                    enabled = is_enable(tuto_f.content)\n",
    "                    trainings[topic_n][tuto_n]['zenodo'] = has_zenodo(tuto_f.content)\n",
    "                elif tuto_f.name == 'slides.html':\n",
    "                    enabled = is_enable(tuto_f.content)\n",
    "                elif tuto_f.name == 'workflows':\n",
    "                    trainings[topic_n][tuto_n]['workflows'] = True\n",
    "            # aggregate stats if enable tutorial\n",
    "            if enabled:\n",
    "                topic_stat['tutorials'] += 1\n",
    "                for c in topic_stat:\n",
    "                    if c == 'tutorials' or c == 'introduction':\n",
    "                        continue\n",
    "                    if trainings[topic_n][tuto_n][c]:\n",
    "                        topic_stat[c] += 1\n",
    "        # add stats in table\n",
    "        stats.loc[topic_t] = topic_stat\n",
    "    # rename table columns\n",
    "    stats = stats.rename(columns = {\n",
    "        'introduction': 'Introduction slide decks',\n",
    "        'tutorials': 'Tutorials',\n",
    "        'workflows': 'Workflows',\n",
    "        'zenodo': 'Data on Zenodo'})\n",
    "    return trainings, stats    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2017-06-01 00:00:00\n",
      "df3ebd4adad9adfced9dd9362a6a8d54cbffcc6b\n",
      "admin\n",
      "database-schema\n",
      "dev-to-production\n",
      "galaxy-docker\n",
      "assembly\n",
      "debruijn-graph-assembly\n",
      "general-introduction\n",
      "chip-seq\n",
      "tal1-binding-site-identification\n",
      "dev\n",
      "bioblend-api\n",
      "data-source-integration\n",
      "github-contribution\n",
      "interactive-environment\n",
      "interactive-tour\n",
      "tool-integration\n",
      "toolshed\n",
      "visualization\n",
      "webhook\n",
      "epigenetics\n",
      "methylation-seq\n",
      "introduction\n",
      "galaxy-intro-101\n",
      "galaxy-intro-collections\n",
      "galaxy-intro-get-data\n",
      "galaxy-intro-peaks2genes\n",
      "galaxy-intro-vis\n",
      "igv-introduction\n",
      "options-for-using-galaxy\n",
      "processing-many-samples-at-once\n",
      "metagenomics\n",
      "mothur-miseq-sop\n",
      "proteomics\n",
      "database-handling\n",
      "labelfree-vs-labelled\n",
      "ntails\n",
      "protein-id-sg-ps\n",
      "protein-quant-sil\n",
      "secretome-prediction\n",
      "sequence-analysis\n",
      "de-novo-rad-seq\n",
      "genetic-map-rad-seq\n",
      "genome-annotation\n",
      "mapping\n",
      "quality-control\n",
      "ref-based-rad-seq\n",
      "training\n",
      "create-new-tutorial\n",
      "good-practices\n",
      "transcriptomics\n",
      "de-novo\n",
      "ref-based\n",
      "variant-analysis\n",
      "diploid-variant-calling\n",
      "exome-seq\n",
      "\n",
      "2017-07-01 00:00:00\n",
      "9bb8d1326fcd742f4292b2c914dca5f5df36923d\n",
      "admin\n",
      "advanced-galaxy-customisation\n",
      "database-schema\n",
      "dev-to-production\n",
      "galaxy-docker\n",
      "assembly\n",
      "debruijn-graph-assembly\n",
      "general-introduction\n",
      "chip-seq\n",
      "chip-seq\n",
      "tal1-binding-site-identification\n",
      "dev\n",
      "architecture\n",
      "bioblend-api\n",
      "conda\n",
      "containers\n",
      "data-source-integration\n",
      "github-contribution\n",
      "interactive-environments\n",
      "interactive-tours\n",
      "tool-integration\n",
      "toolshed\n",
      "visualization-charts\n",
      "visualization-generic\n",
      "webhooks\n",
      "epigenetics\n",
      "methylation-seq\n",
      "introduction\n",
      "galaxy-intro-101\n",
      "galaxy-intro-collections\n",
      "galaxy-intro-get-data\n",
      "galaxy-intro-peaks2genes\n",
      "galaxy-intro-vis\n",
      "igv-introduction\n",
      "options-for-using-galaxy\n",
      "processing-many-samples-at-once\n",
      "metagenomics\n",
      "general-tutorial\n",
      "mothur-miseq-sop\n",
      "proteomics\n",
      "database-handling\n",
      "labelfree-vs-labelled\n",
      "metaproteomics\n",
      "ntails\n",
      "protein-id-sg-ps\n",
      "protein-quant-sil\n",
      "secretome-prediction\n",
      "sequence-analysis\n",
      "de-novo-rad-seq\n",
      "genetic-map-rad-seq\n",
      "genome-annotation\n",
      "mapping\n",
      "quality-control\n",
      "ref-based-rad-seq\n",
      "srna\n",
      "training\n",
      "create-new-tutorial-content\n",
      "create-new-tutorial-docker\n",
      "create-new-tutorial-jekyll\n",
      "create-new-tutorial-metadata\n",
      "create-new-tutorial-tours\n",
      "good-practices\n",
      "transcriptomics\n",
      "de-novo\n",
      "ref-based\n",
      "srna\n",
      "usegalaxy\n",
      "collections\n",
      "dip\n",
      "dunovo\n",
      "history\n",
      "ngs\n",
      "non-dip\n",
      "rb-rnaseq\n",
      "variant-analysis\n",
      "diploid-variant-calling\n",
      "exome-seq\n",
      "\n",
      "2017-08-01 00:00:00\n",
      "ce44f1e3f8d774d50ede5582ac420bb6bfe9aca4\n",
      "admin\n",
      "advanced-galaxy-customisation\n",
      "database-schema\n",
      "dev-to-production\n",
      "galaxy-docker\n",
      "assembly\n",
      "debruijn-graph-assembly\n",
      "general-introduction\n",
      "chip-seq\n",
      "chip-seq\n",
      "tal1-binding-site-identification\n",
      "dev\n",
      "architecture\n",
      "bioblend-api\n",
      "conda\n",
      "containers\n",
      "data-source-integration\n",
      "github-contribution\n",
      "interactive-environments\n",
      "interactive-tours\n",
      "tool-integration\n",
      "toolshed\n",
      "visualization-charts\n",
      "visualization-generic\n",
      "webhooks\n",
      "epigenetics\n",
      "methylation-seq\n",
      "introduction\n",
      "galaxy-intro-101\n",
      "galaxy-intro-collections\n",
      "galaxy-intro-get-data\n",
      "galaxy-intro-peaks2genes\n",
      "galaxy-intro-vis\n",
      "igv-introduction\n",
      "options-for-using-galaxy\n",
      "processing-many-samples-at-once\n",
      "metagenomics\n",
      "general-tutorial\n",
      "mothur-miseq-sop\n",
      "proteomics\n",
      "database-handling\n",
      "labelfree-vs-labelled\n",
      "metaproteomics\n",
      "ntails\n",
      "protein-id-sg-ps\n",
      "protein-quant-sil\n",
      "secretome-prediction\n",
      "sequence-analysis\n",
      "de-novo-rad-seq\n",
      "genetic-map-rad-seq\n",
      "genome-annotation\n",
      "mapping\n",
      "quality-control\n",
      "ref-based-rad-seq\n",
      "training\n",
      "create-new-tutorial-content\n",
      "create-new-tutorial-docker\n",
      "create-new-tutorial-jekyll\n",
      "create-new-tutorial-metadata\n",
      "create-new-tutorial-tours\n",
      "good-practices\n",
      "transcriptomics\n",
      "de-novo\n",
      "ref-based\n",
      "srna\n",
      "usegalaxy\n",
      "collections\n",
      "dip\n",
      "dunovo\n",
      "history\n",
      "ngs\n",
      "non-dip\n",
      "rb-rnaseq\n",
      "variant-analysis\n",
      "diploid-variant-calling\n",
      "exome-seq\n",
      "\n",
      "2017-09-01 00:00:00\n",
      "bd5ff82eba8bd58e41fa4f4b89d191358ff55945\n",
      "admin\n",
      "advanced-galaxy-customisation\n",
      "database-schema\n",
      "dev-to-production\n",
      "galaxy-docker\n",
      "assembly\n",
      "debruijn-graph-assembly\n",
      "general-introduction\n",
      "chip-seq\n",
      "chip-seq\n",
      "tal1-binding-site-identification\n",
      "dev\n",
      "architecture\n",
      "bioblend-api\n",
      "conda\n",
      "containers\n",
      "data-source-integration\n",
      "github-contribution\n",
      "interactive-environments\n",
      "interactive-tours\n",
      "tool-integration\n",
      "toolshed\n",
      "visualization-charts\n",
      "visualization-generic\n",
      "webhooks\n",
      "epigenetics\n",
      "methylation-seq\n",
      "introduction\n",
      "galaxy-intro-101\n",
      "galaxy-intro-collections\n",
      "galaxy-intro-get-data\n",
      "galaxy-intro-peaks2genes\n",
      "galaxy-intro-vis\n",
      "igv-introduction\n",
      "options-for-using-galaxy\n",
      "processing-many-samples-at-once\n",
      "metagenomics\n",
      "general-tutorial\n",
      "mothur-miseq-sop\n",
      "proteomics\n",
      "database-handling\n",
      "labelfree-vs-labelled\n",
      "metaproteomics\n",
      "ntails\n",
      "protein-id-sg-ps\n",
      "protein-quant-sil\n",
      "secretome-prediction\n",
      "sequence-analysis\n",
      "de-novo-rad-seq\n",
      "genetic-map-rad-seq\n",
      "genome-annotation\n",
      "mapping\n",
      "quality-control\n",
      "ref-based-rad-seq\n",
      "training\n",
      "create-new-tutorial-content\n",
      "create-new-tutorial-docker\n",
      "create-new-tutorial-jekyll\n",
      "create-new-tutorial-metadata\n",
      "create-new-tutorial-tours\n",
      "good-practices\n",
      "transcriptomics\n",
      "de-novo\n",
      "ref-based\n",
      "srna\n",
      "usegalaxy\n",
      "collections\n",
      "dip\n",
      "dunovo\n",
      "history\n",
      "ngs\n",
      "non-dip\n",
      "rb-rnaseq\n",
      "variant-analysis\n",
      "diploid-variant-calling\n",
      "exome-seq\n",
      "\n",
      "2017-10-01 00:00:00\n",
      "5e7be35665ca41f14dd2c8750390f3ae71e8cbb8\n",
      "admin\n",
      "advanced-galaxy-customisation\n",
      "database-schema\n",
      "dev-to-production\n",
      "galaxy-docker\n",
      "assembly\n",
      "debruijn-graph-assembly\n",
      "general-introduction\n",
      "chip-seq\n",
      "estrogen-receptor-binding-site-identification\n",
      "tal1-binding-site-identification\n",
      "dev\n",
      "architecture\n",
      "bioblend-api\n",
      "conda\n",
      "containers\n",
      "data-source-integration\n",
      "github-contribution\n",
      "interactive-environments\n",
      "interactive-tours\n",
      "tool-integration\n",
      "toolshed\n",
      "visualization-charts\n",
      "visualization-generic\n",
      "webhooks\n",
      "epigenetics\n",
      "methylation-seq\n",
      "introduction\n",
      "galaxy-intro-101\n",
      "galaxy-intro-collections\n",
      "galaxy-intro-get-data\n",
      "galaxy-intro-peaks2genes\n",
      "galaxy-intro-vis\n",
      "igv-introduction\n",
      "options-for-using-galaxy\n",
      "processing-many-samples-at-once\n",
      "metagenomics\n",
      "general-tutorial\n",
      "mothur-miseq-sop\n",
      "proteomics\n",
      "database-handling\n",
      "labelfree-vs-labelled\n",
      "metaproteomics\n",
      "ntails\n",
      "protein-id-sg-ps\n",
      "protein-quant-sil\n",
      "secretome-prediction\n",
      "sequence-analysis\n",
      "de-novo-rad-seq\n",
      "genetic-map-rad-seq\n",
      "genome-annotation\n",
      "mapping\n",
      "quality-control\n",
      "ref-based-rad-seq\n",
      "training\n",
      "create-new-tutorial-content\n",
      "create-new-tutorial-docker\n",
      "create-new-tutorial-jekyll\n",
      "create-new-tutorial-metadata\n",
      "create-new-tutorial-tours\n",
      "good-practices\n",
      "transcriptomics\n",
      "de-novo\n",
      "ref-based\n",
      "srna\n",
      "usegalaxy\n",
      "collections\n",
      "dip\n",
      "dunovo\n",
      "history\n",
      "ngs\n",
      "non-dip\n",
      "rb-rnaseq\n",
      "variant-analysis\n",
      "diploid-variant-calling\n",
      "exome-seq\n",
      "\n",
      "2017-11-01 00:00:00\n",
      "65acb12bb77375bf6509b5ccfaf229b16dea4085\n",
      "admin\n",
      "advanced-galaxy-customisation\n",
      "database-schema\n",
      "dev-to-production\n",
      "galaxy-docker\n",
      "assembly\n",
      "debruijn-graph-assembly\n",
      "general-introduction\n",
      "unicycler-assembly\n",
      "chip-seq\n",
      "estrogen-receptor-binding-site-identification\n",
      "tal1-binding-site-identification\n",
      "dev\n",
      "architecture\n",
      "bioblend-api\n",
      "conda\n",
      "containers\n",
      "data-source-integration\n",
      "github-contribution\n",
      "interactive-environments\n",
      "interactive-tours\n",
      "tool-integration\n",
      "toolshed\n",
      "visualization-charts\n",
      "visualization-generic\n",
      "webhooks\n",
      "epigenetics\n",
      "methylation-seq\n",
      "introduction\n",
      "galaxy-intro-101\n",
      "galaxy-intro-collections\n",
      "galaxy-intro-get-data\n",
      "galaxy-intro-history-to-workflow\n",
      "galaxy-intro-peaks2genes\n",
      "galaxy-intro-strands\n",
      "galaxy-intro-vis\n",
      "igv-introduction\n",
      "options-for-using-galaxy\n",
      "processing-many-samples-at-once\n",
      "metagenomics\n",
      "general-tutorial\n",
      "mothur-miseq-sop\n",
      "proteomics\n",
      "database-handling\n",
      "labelfree-vs-labelled\n",
      "metaproteomics\n",
      "ntails\n",
      "protein-id-oms\n",
      "protein-id-sg-ps\n",
      "protein-quant-sil\n",
      "secretome-prediction\n",
      "sequence-analysis\n",
      "de-novo-rad-seq\n",
      "genetic-map-rad-seq\n",
      "genome-annotation\n",
      "mapping\n",
      "quality-control\n",
      "ref-based-rad-seq\n",
      "training\n",
      "create-new-tutorial-content\n",
      "create-new-tutorial-docker\n",
      "create-new-tutorial-jekyll\n",
      "create-new-tutorial-metadata\n",
      "create-new-tutorial-tours\n",
      "good-practices\n",
      "transcriptomics\n",
      "de-novo\n",
      "ref-based\n",
      "rna-seq-viz-with-cummerbund\n",
      "srna\n",
      "usegalaxy\n",
      "collections\n",
      "dip\n",
      "dunovo\n",
      "history\n",
      "ngs\n",
      "non-dip\n",
      "rb-rnaseq\n",
      "variant-analysis\n",
      "diploid-variant-calling\n",
      "exome-seq\n",
      "\n",
      "2017-12-01 00:00:00\n",
      "e921974df6aa0c78e8e4d42d3f0491bfbcc2a2e0\n",
      "admin\n",
      "advanced-galaxy-customisation\n",
      "database-schema\n",
      "dev-to-production\n",
      "galaxy-docker\n",
      "assembly\n",
      "debruijn-graph-assembly\n",
      "general-introduction\n",
      "unicycler-assembly\n",
      "chip-seq\n",
      "estrogen-receptor-binding-site-identification\n",
      "tal1-binding-site-identification\n",
      "dev\n",
      "architecture\n",
      "bioblend-api\n",
      "conda\n",
      "containers\n",
      "data-source-integration\n",
      "github-contribution\n",
      "interactive-environments\n",
      "interactive-tours\n",
      "tool-integration\n",
      "toolshed\n",
      "visualization-charts\n",
      "visualization-generic\n",
      "webhooks\n",
      "epigenetics\n",
      "methylation-seq\n",
      "introduction\n",
      "galaxy-intro-101\n",
      "galaxy-intro-collections\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "galaxy-intro-get-data\n",
      "galaxy-intro-history-to-workflow\n",
      "galaxy-intro-peaks2genes\n",
      "galaxy-intro-strands\n",
      "galaxy-intro-vis\n",
      "igv-introduction\n",
      "options-for-using-galaxy\n",
      "processing-many-samples-at-once\n",
      "metagenomics\n",
      "general-tutorial\n",
      "mothur-miseq-sop\n",
      "proteomics\n",
      "database-handling\n",
      "labelfree-vs-labelled\n",
      "metaproteomics\n",
      "ntails\n",
      "protein-id-oms\n",
      "protein-id-sg-ps\n",
      "protein-quant-sil\n",
      "secretome-prediction\n",
      "sequence-analysis\n",
      "de-novo-rad-seq\n",
      "genetic-map-rad-seq\n",
      "genome-annotation\n",
      "mapping\n",
      "quality-control\n",
      "ref-based-rad-seq\n",
      "training\n",
      "create-new-tutorial-content\n",
      "create-new-tutorial-docker\n",
      "create-new-tutorial-jekyll\n",
      "create-new-tutorial-metadata\n",
      "create-new-tutorial-tours\n",
      "good-practices\n",
      "transcriptomics\n",
      "de-novo\n",
      "ref-based\n",
      "rna-seq-viz-with-cummerbund\n",
      "srna\n",
      "usegalaxy\n",
      "collections\n",
      "dip\n",
      "dunovo\n",
      "history\n",
      "ngs\n",
      "non-dip\n",
      "rb-rnaseq\n",
      "variant-analysis\n",
      "diploid-variant-calling\n",
      "exome-seq\n",
      "\n",
      "2018-01-01 00:00:00\n",
      "cd17337bde871053716627a5cc241c334b0e5ca6\n",
      "admin\n",
      "advanced-galaxy-customisation\n",
      "database-schema\n",
      "dev-to-production\n",
      "galaxy-docker\n",
      "monitoring-maintenance\n",
      "assembly\n",
      "debruijn-graph-assembly\n",
      "general-introduction\n",
      "unicycler-assembly\n",
      "chip-seq\n",
      "estrogen-receptor-binding-site-identification\n",
      "tal1-binding-site-identification\n",
      "dev\n",
      "architecture\n",
      "bioblend-api\n",
      "conda\n",
      "containers\n",
      "data-source-integration\n",
      "github-contribution\n",
      "interactive-environments\n",
      "interactive-tours\n",
      "tool-integration\n",
      "toolshed\n",
      "visualization-charts\n",
      "visualization-generic\n",
      "webhooks\n",
      "epigenetics\n",
      "methylation-seq\n",
      "introduction\n",
      "galaxy-intro-101\n",
      "galaxy-intro-collections\n",
      "galaxy-intro-get-data\n",
      "galaxy-intro-history-to-workflow\n",
      "galaxy-intro-peaks2genes\n",
      "galaxy-intro-strands\n",
      "galaxy-intro-vis\n",
      "igv-introduction\n",
      "options-for-using-galaxy\n",
      "processing-many-samples-at-once\n",
      "metagenomics\n",
      "general-tutorial\n",
      "mothur-miseq-sop\n",
      "proteomics\n",
      "database-handling\n",
      "labelfree-vs-labelled\n",
      "metaproteomics\n",
      "ntails\n",
      "protein-id-oms\n",
      "protein-id-sg-ps\n",
      "protein-quant-sil\n",
      "secretome-prediction\n",
      "sequence-analysis\n",
      "de-novo-rad-seq\n",
      "genetic-map-rad-seq\n",
      "genome-annotation\n",
      "mapping\n",
      "quality-control\n",
      "ref-based-rad-seq\n",
      "training\n",
      "create-new-tutorial-content\n",
      "create-new-tutorial-docker\n",
      "create-new-tutorial-jekyll\n",
      "create-new-tutorial-metadata\n",
      "create-new-tutorial-tours\n",
      "good-practices\n",
      "transcriptomics\n",
      "de-novo\n",
      "ref-based\n",
      "rna-seq-viz-with-cummerbund\n",
      "srna\n",
      "usegalaxy\n",
      "collections\n",
      "dip\n",
      "dunovo\n",
      "history\n",
      "ngs\n",
      "non-dip\n",
      "rb-rnaseq\n",
      "variant-analysis\n",
      "diploid-variant-calling\n",
      "exome-seq\n",
      "\n",
      "2018-02-01 00:00:00\n",
      "a2a1552ecabd24bace8a395b7fa38896e5824316\n",
      "admin\n",
      "advanced-galaxy-customisation\n",
      "connect-to-compute-cluster\n",
      "database-schema\n",
      "deployment-platforms-options\n",
      "dev-to-production\n",
      "galaxy-docker\n",
      "monitoring-maintenance\n",
      "users-groups-quotas\n",
      "assembly\n",
      "debruijn-graph-assembly\n",
      "general-introduction\n",
      "unicycler-assembly\n",
      "chip-seq\n",
      "estrogen-receptor-binding-site-identification\n",
      "tal1-binding-site-identification\n",
      "dev\n",
      "architecture\n",
      "bioblend-api\n",
      "conda\n",
      "containers\n",
      "data-source-integration\n",
      "github-contribution\n",
      "interactive-environments\n",
      "interactive-tours\n",
      "tool-integration\n",
      "toolshed\n",
      "visualization-charts\n",
      "visualization-generic\n",
      "webhooks\n",
      "epigenetics\n",
      "methylation-seq\n",
      "introduction\n",
      "galaxy-intro-101\n",
      "galaxy-intro-collections\n",
      "galaxy-intro-get-data\n",
      "galaxy-intro-history-to-workflow\n",
      "galaxy-intro-history\n",
      "galaxy-intro-ngs-data-managment\n",
      "galaxy-intro-peaks2genes\n",
      "galaxy-intro-strands\n",
      "galaxy-intro-vis\n",
      "igv-introduction\n",
      "options-for-using-galaxy\n",
      "processing-many-samples-at-once\n",
      "metagenomics\n",
      "general-tutorial\n",
      "mothur-miseq-sop\n",
      "proteomics\n",
      "database-handling\n",
      "labelfree-vs-labelled\n",
      "metaproteomics\n",
      "ntails\n",
      "protein-id-oms\n",
      "protein-id-sg-ps\n",
      "protein-quant-sil\n",
      "secretome-prediction\n",
      "sequence-analysis\n",
      "de-novo-rad-seq\n",
      "genetic-map-rad-seq\n",
      "genome-annotation\n",
      "mapping\n",
      "quality-control\n",
      "ref-based-rad-seq\n",
      "training\n",
      "create-new-tutorial-content\n",
      "create-new-tutorial-docker\n",
      "create-new-tutorial-jekyll\n",
      "create-new-tutorial-metadata\n",
      "create-new-tutorial-tours\n",
      "good-practices\n",
      "transcriptomics\n",
      "de-novo\n",
      "rb-rnaseq\n",
      "ref-based\n",
      "rna-seq-viz-with-cummerbund\n",
      "srna\n",
      "variant-analysis\n",
      "dip\n",
      "diploid-variant-calling\n",
      "dunovo\n",
      "exome-seq\n",
      "non-dip\n",
      "\n",
      "2018-03-01 00:00:00\n",
      "51c8e4d5c706f26549105a09f3f67b33caad6ace\n",
      "admin\n",
      "advanced-galaxy-customisation\n",
      "connect-to-compute-cluster\n",
      "database-schema\n",
      "deployment-platforms-options\n",
      "dev-to-production\n",
      "galaxy-docker\n",
      "monitoring-maintenance\n",
      "users-groups-quotas\n",
      "assembly\n",
      "debruijn-graph-assembly\n",
      "general-introduction\n",
      "unicycler-assembly\n",
      "chip-seq\n",
      "estrogen-receptor-binding-site-identification\n",
      "tal1-binding-site-identification\n",
      "dev\n",
      "architecture\n",
      "bioblend-api\n",
      "conda\n",
      "conda_sys\n",
      "containers\n",
      "data-source-integration\n",
      "github-contribution\n",
      "interactive-environments\n",
      "interactive-tours\n",
      "tool-integration\n",
      "toolshed\n",
      "visualization-charts\n",
      "visualization-generic\n",
      "webhooks\n",
      "epigenetics\n",
      "hicexplorer\n",
      "methylation-seq\n",
      "introduction\n",
      "galaxy-intro-101\n",
      "galaxy-intro-collections\n",
      "galaxy-intro-get-data\n",
      "galaxy-intro-history-to-workflow\n",
      "galaxy-intro-history\n",
      "galaxy-intro-ngs-data-managment\n",
      "galaxy-intro-peaks2genes\n",
      "galaxy-intro-strands\n",
      "galaxy-intro-vis\n",
      "igv-introduction\n",
      "options-for-using-galaxy\n",
      "processing-many-samples-at-once\n",
      "metagenomics\n",
      "general-tutorial\n",
      "mothur-miseq-sop\n",
      "proteomics\n",
      "database-handling\n",
      "labelfree-vs-labelled\n",
      "metaproteomics\n",
      "ntails\n",
      "protein-id-oms\n",
      "protein-id-sg-ps\n",
      "protein-quant-sil\n",
      "secretome-prediction\n",
      "sequence-analysis\n",
      "de-novo-rad-seq\n",
      "genetic-map-rad-seq\n",
      "genome-annotation\n",
      "mapping\n",
      "quality-control\n",
      "ref-based-rad-seq\n",
      "training\n",
      "create-new-tutorial-content\n",
      "create-new-tutorial-docker\n",
      "create-new-tutorial-jekyll\n",
      "create-new-tutorial-metadata\n",
      "create-new-tutorial-tours\n",
      "good-practices\n",
      "transcriptomics\n",
      "de-novo\n",
      "rb-rnaseq\n",
      "ref-based\n",
      "rna-seq-viz-with-cummerbund\n",
      "srna\n",
      "variant-analysis\n",
      "dip\n",
      "diploid-variant-calling\n",
      "dunovo\n",
      "exome-seq\n",
      "microbial-variants\n",
      "non-dip\n",
      "\n",
      "2018-04-01 00:00:00\n",
      "385ee9e46bb7d90183be476a7a68e3bec4309658\n",
      "admin\n"
     ]
    },
    {
     "ename": "RateLimitExceededException",
     "evalue": "403 {\"message\": \"API rate limit exceeded for user ID 1842467.\", \"documentation_url\": \"https://docs.github.com/rest/overview/resources-in-the-rest-api#rate-limiting\"}",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mRateLimitExceededException\u001b[0m                Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-82-832721520244>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      6\u001b[0m     \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0md\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      7\u001b[0m     \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mgit_com\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 8\u001b[0;31m     \u001b[0mtraining_contents\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0md\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mstats_per_date\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0md\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mextract_training_content\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mgit_com\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      9\u001b[0m     aggr_stats = (stats_per_date[d]\n\u001b[1;32m     10\u001b[0m                   \u001b[0;34m.\u001b[0m\u001b[0msum\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m<ipython-input-81-3e4e0fc3632f>\u001b[0m in \u001b[0;36mextract_training_content\u001b[0;34m(ref)\u001b[0m\n\u001b[1;32m     63\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mtopic_c\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mname\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m'slides'\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     64\u001b[0m                 topic_stat['introduction'] += len(\n\u001b[0;32m---> 65\u001b[0;31m                     repo.get_contents(\"%s/slides/\" % topic.path, ref))\n\u001b[0m\u001b[1;32m     66\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mtopic_c\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mname\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m'tutorials'\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     67\u001b[0m                 \u001b[0mhas_tuto_folder\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mTrue\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/envs/gtn-community-paper-2020/lib/python3.6/site-packages/github/Repository.py\u001b[0m in \u001b[0;36mget_contents\u001b[0;34m(self, path, ref)\u001b[0m\n\u001b[1;32m   1753\u001b[0m             \u001b[0;34m\"GET\"\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1754\u001b[0m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0murl\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0;34m\"/contents/\"\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0murllib\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mparse\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mquote\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mpath\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1755\u001b[0;31m             \u001b[0mparameters\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0murl_parameters\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1756\u001b[0m         )\n\u001b[1;32m   1757\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/envs/gtn-community-paper-2020/lib/python3.6/site-packages/github/Requester.py\u001b[0m in \u001b[0;36mrequestJsonAndCheck\u001b[0;34m(self, verb, url, parameters, headers, input)\u001b[0m\n\u001b[1;32m    317\u001b[0m         return self.__check(\n\u001b[1;32m    318\u001b[0m             *self.requestJson(\n\u001b[0;32m--> 319\u001b[0;31m                 \u001b[0mverb\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0murl\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mparameters\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mheaders\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0minput\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__customConnection\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0murl\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    320\u001b[0m             )\n\u001b[1;32m    321\u001b[0m         )\n",
      "\u001b[0;32m~/miniconda3/envs/gtn-community-paper-2020/lib/python3.6/site-packages/github/Requester.py\u001b[0m in \u001b[0;36m__check\u001b[0;34m(self, status, responseHeaders, output)\u001b[0m\n\u001b[1;32m    340\u001b[0m         \u001b[0moutput\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__structuredFromJson\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0moutput\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    341\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mstatus\u001b[0m \u001b[0;34m>=\u001b[0m \u001b[0;36m400\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 342\u001b[0;31m             \u001b[0;32mraise\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__createException\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mstatus\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mresponseHeaders\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0moutput\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    343\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mresponseHeaders\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0moutput\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    344\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mRateLimitExceededException\u001b[0m: 403 {\"message\": \"API rate limit exceeded for user ID 1842467.\", \"documentation_url\": \"https://docs.github.com/rest/overview/resources-in-the-rest-api#rate-limiting\"}"
     ]
    }
   ],
   "source": [
    "# extract training content\n",
    "training_contents = {}\n",
    "stats_per_date = {}\n",
    "global_stats = pd.DataFrame(columns=[], index=[])\n",
    "for d, git_com in commits.items():\n",
    "    print(d)\n",
    "    print(git_com)\n",
    "    training_contents[d], stats_per_date[d] = extract_training_content(git_com)\n",
    "    aggr_stats = (stats_per_date[d]\n",
    "                  .sum()\n",
    "                  .add(pd.Series([len(training_contents[d])], index=['Topics']), fill_value=0)\n",
    "                  .to_frame(name=d))\n",
    "    global_stats = pd.concat([global_stats, aggr_stats], axis=1)\n",
    "    print()\n",
    "# format table\n",
    "global_stats = (global_stats\n",
    "    .reindex(['Topics', 'Introduction slide decks', 'Tutorials', 'Workflows', 'Data on Zenodo'])\n",
    "    .astype(int))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Topics, number of tutorials and technical support (today)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Introduction slide decks</th>\n",
       "      <th>Tutorials</th>\n",
       "      <th>Slide decks</th>\n",
       "      <th>Hands-on tutorials</th>\n",
       "      <th>Workflows</th>\n",
       "      <th>Data on Zenodo</th>\n",
       "      <th>Data on data library</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Galaxy Server administration</th>\n",
       "      <td>1</td>\n",
       "      <td>34</td>\n",
       "      <td>26</td>\n",
       "      <td>19</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembly</th>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Climate</th>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Computational chemistry</th>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>5</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Contributing to the Galaxy Training Material</th>\n",
       "      <td>1</td>\n",
       "      <td>10</td>\n",
       "      <td>2</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Development in Galaxy</th>\n",
       "      <td>1</td>\n",
       "      <td>13</td>\n",
       "      <td>12</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Ecology</th>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Epigenetics</th>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>4</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>User Interface and Data Manipulation</th>\n",
       "      <td>0</td>\n",
       "      <td>16</td>\n",
       "      <td>1</td>\n",
       "      <td>15</td>\n",
       "      <td>2</td>\n",
       "      <td>15</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genome Annotation</th>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Imaging</th>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Teaching and Hosting Galaxy training</th>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Introduction to Galaxy Analyses</th>\n",
       "      <td>1</td>\n",
       "      <td>10</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>3</td>\n",
       "      <td>9</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Metabolomics</th>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Metagenomics</th>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Proteomics</th>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "      <td>14</td>\n",
       "      <td>15</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sequence analysis</th>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Statistics and machine learning</th>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Transcriptomics</th>\n",
       "      <td>1</td>\n",
       "      <td>23</td>\n",
       "      <td>6</td>\n",
       "      <td>22</td>\n",
       "      <td>18</td>\n",
       "      <td>22</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Variant Analysis</th>\n",
       "      <td>1</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Visualisation</th>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                             Introduction slide decks  \\\n",
       "Galaxy Server administration                                        1   \n",
       "Assembly                                                            0   \n",
       "Climate                                                             0   \n",
       "Computational chemistry                                             0   \n",
       "Contributing to the Galaxy Training Material                        1   \n",
       "Development in Galaxy                                               1   \n",
       "Ecology                                                             0   \n",
       "Epigenetics                                                         2   \n",
       "User Interface and Data Manipulation                                0   \n",
       "Genome Annotation                                                   1   \n",
       "Imaging                                                             0   \n",
       "Teaching and Hosting Galaxy training                                0   \n",
       "Introduction to Galaxy Analyses                                     1   \n",
       "Metabolomics                                                        1   \n",
       "Metagenomics                                                        1   \n",
       "Proteomics                                                          0   \n",
       "Sequence analysis                                                   0   \n",
       "Statistics and machine learning                                     0   \n",
       "Transcriptomics                                                     1   \n",
       "Variant Analysis                                                    1   \n",
       "Visualisation                                                       1   \n",
       "\n",
       "                                             Tutorials Slide decks  \\\n",
       "Galaxy Server administration                        34          26   \n",
       "Assembly                                             5           3   \n",
       "Climate                                              2           0   \n",
       "Computational chemistry                              6           0   \n",
       "Contributing to the Galaxy Training Material        10           2   \n",
       "Development in Galaxy                               13          12   \n",
       "Ecology                                              5           0   \n",
       "Epigenetics                                          5           2   \n",
       "User Interface and Data Manipulation                16           1   \n",
       "Genome Annotation                                    3           2   \n",
       "Imaging                                              3           1   \n",
       "Teaching and Hosting Galaxy training                 5           1   \n",
       "Introduction to Galaxy Analyses                     10           1   \n",
       "Metabolomics                                         4           1   \n",
       "Metagenomics                                         6           1   \n",
       "Proteomics                                          15           0   \n",
       "Sequence analysis                                    2           2   \n",
       "Statistics and machine learning                      8           0   \n",
       "Transcriptomics                                     23           6   \n",
       "Variant Analysis                                     8           0   \n",
       "Visualisation                                        2           2   \n",
       "\n",
       "                                             Hands-on tutorials Workflows  \\\n",
       "Galaxy Server administration                                 19         0   \n",
       "Assembly                                                      5         5   \n",
       "Climate                                                       2         0   \n",
       "Computational chemistry                                       6         5   \n",
       "Contributing to the Galaxy Training Material                  9         1   \n",
       "Development in Galaxy                                         4         0   \n",
       "Ecology                                                       5         5   \n",
       "Epigenetics                                                   5         4   \n",
       "User Interface and Data Manipulation                         15         2   \n",
       "Genome Annotation                                             3         2   \n",
       "Imaging                                                       3         3   \n",
       "Teaching and Hosting Galaxy training                          4         0   \n",
       "Introduction to Galaxy Analyses                               9         3   \n",
       "Metabolomics                                                  4         3   \n",
       "Metagenomics                                                  6         6   \n",
       "Proteomics                                                   15        14   \n",
       "Sequence analysis                                             2         2   \n",
       "Statistics and machine learning                               8         7   \n",
       "Transcriptomics                                              22        18   \n",
       "Variant Analysis                                              8         7   \n",
       "Visualisation                                                 2         1   \n",
       "\n",
       "                                             Data on Zenodo  \\\n",
       "Galaxy Server administration                              9   \n",
       "Assembly                                                  5   \n",
       "Climate                                                   2   \n",
       "Computational chemistry                                   6   \n",
       "Contributing to the Galaxy Training Material              3   \n",
       "Development in Galaxy                                     0   \n",
       "Ecology                                                   4   \n",
       "Epigenetics                                               5   \n",
       "User Interface and Data Manipulation                     15   \n",
       "Genome Annotation                                         3   \n",
       "Imaging                                                   3   \n",
       "Teaching and Hosting Galaxy training                      0   \n",
       "Introduction to Galaxy Analyses                           9   \n",
       "Metabolomics                                              4   \n",
       "Metagenomics                                              6   \n",
       "Proteomics                                               15   \n",
       "Sequence analysis                                         2   \n",
       "Statistics and machine learning                           8   \n",
       "Transcriptomics                                          22   \n",
       "Variant Analysis                                          8   \n",
       "Visualisation                                             2   \n",
       "\n",
       "                                             Data on data library  \n",
       "Galaxy Server administration                                    0  \n",
       "Assembly                                                        4  \n",
       "Climate                                                         0  \n",
       "Computational chemistry                                         0  \n",
       "Contributing to the Galaxy Training Material                    0  \n",
       "Development in Galaxy                                           0  \n",
       "Ecology                                                         4  \n",
       "Epigenetics                                                     5  \n",
       "User Interface and Data Manipulation                            1  \n",
       "Genome Annotation                                               3  \n",
       "Imaging                                                         2  \n",
       "Teaching and Hosting Galaxy training                            0  \n",
       "Introduction to Galaxy Analyses                                 3  \n",
       "Metabolomics                                                    3  \n",
       "Metagenomics                                                    4  \n",
       "Proteomics                                                     10  \n",
       "Sequence analysis                                               2  \n",
       "Statistics and machine learning                                 8  \n",
       "Transcriptomics                                                16  \n",
       "Variant Analysis                                                7  \n",
       "Visualisation                                                   2  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "stats_per_date['today']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Global informations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>22.06.17</th>\n",
       "      <th>22.06.18</th>\n",
       "      <th>22.06.19</th>\n",
       "      <th>22.06.20</th>\n",
       "      <th>today</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Topics</th>\n",
       "      <td>13</td>\n",
       "      <td>14</td>\n",
       "      <td>19</td>\n",
       "      <td>22</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Introduction slide decks</th>\n",
       "      <td>13</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Tutorials</th>\n",
       "      <td>63</td>\n",
       "      <td>86</td>\n",
       "      <td>134</td>\n",
       "      <td>177</td>\n",
       "      <td>185</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Slide decks</th>\n",
       "      <td>21</td>\n",
       "      <td>31</td>\n",
       "      <td>51</td>\n",
       "      <td>60</td>\n",
       "      <td>63</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Hands-on tutorials</th>\n",
       "      <td>49</td>\n",
       "      <td>68</td>\n",
       "      <td>105</td>\n",
       "      <td>148</td>\n",
       "      <td>156</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Workflows</th>\n",
       "      <td>23</td>\n",
       "      <td>32</td>\n",
       "      <td>57</td>\n",
       "      <td>85</td>\n",
       "      <td>88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Data on Zenodo</th>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>86</td>\n",
       "      <td>123</td>\n",
       "      <td>131</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Data on data library</th>\n",
       "      <td>36</td>\n",
       "      <td>38</td>\n",
       "      <td>53</td>\n",
       "      <td>72</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                          22.06.17  22.06.18  22.06.19  22.06.20  today\n",
       "Topics                          13        14        19        22     21\n",
       "Introduction slide decks        13        11        11        12     12\n",
       "Tutorials                       63        86       134       177    185\n",
       "Slide decks                     21        31        51        60     63\n",
       "Hands-on tutorials              49        68       105       148    156\n",
       "Workflows                       23        32        57        85     88\n",
       "Data on Zenodo                   1         2        86       123    131\n",
       "Data on data library            36        38        53        72     74"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "global_stats"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAagAAAEYCAYAAAAJeGK1AAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjMuMiwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy8vihELAAAACXBIWXMAAAsTAAALEwEAmpwYAAAhvUlEQVR4nO3de3xU1b338c8PIsYLF4VIwahRKgUlYiFcTn0O+miR411BUUqPICKlyqPPS0XT1gtaWkAfq63gBR9FVOSmeAErrQrW27ESLchN9IBYo6gBBRMVhPA7f+yddBgSGMhOZs/k+3698urM2re1p4zf2WuvvZa5OyIiInHTJN0VEBERqYkCSkREYkkBJSIisaSAEhGRWFJAiYhILOWkuwI1adOmjRcUFKS7GiIi0gDefvvt9e6el1wey4AqKCigpKQk3dUQEZEGYGYf1VSuJj4REYklBZSIiMSSAkpERGIplvegarJ161ZKS0vZvHlzuquSkXJzc8nPz2efffZJd1VERFKSMQFVWlpK8+bNKSgowMzSXZ2M4u5s2LCB0tJSjjzyyHRXR0QkJRnTxLd582Zat26tcNoLZkbr1q119SkiGSVjAgpQONWBPjsRyTQZFVAiItJ4ZMw9qGQFxc9Fur+148/Y7Toff/wxF198MZ999hlNmjRhxIgRXHXVVYwePZq5c+fSrFkzOnTowJQpU2jVqtVO28+fP5+rrrqKyspKhg8fTnFxcfWyu+++m4kTJ5KTk8MZZ5zBbbfdttP2w4YNY968eRxyyCEsW7asuvzCCy9k1apVAGzcuJFWrVqxePHiPf8QRERiJGMDKh1ycnK444476NatG+Xl5XTv3p2+ffvSt29fxo0bR05ODtdffz3jxo1jwoQJO2xbWVnJFVdcwQsvvEB+fj49evTg7LPP5phjjmHhwoU888wzvPvuu+y777588cUXNR5/6NChjBo1iosvvniH8pkzZ1a/vuaaa2jZsmX0Jy8ikZo0ckFk+7rivpMj21ecqIlvD7Rr145u3boB0Lx5czp37swnn3zCqaeeSk5OkPW9e/emtLR0p23feustfvjDH3LUUUfRrFkzLrroIp555hkA7r33XoqLi9l3330BOOSQQ2o8fp8+fTj44INrrZ+7M2vWLAYNGlSn8xQRiQMF1F5au3Yt//jHP+jVq9cO5Q899BCnnXbaTut/8sknHHbYYdXv8/Pz+eSTTwB4//33efXVV+nVqxcnnngiixYt2qs6vfrqq7Rt25ajjz56r7YXEYkTNfHthYqKCgYMGMBdd91FixYtqst/97vfkZOTw+DBg3faxt13KqvqWbdt2za++uor3nzzTRYtWsTAgQNZs2bNHve8mz59uq6eRCRrKKD20NatWxkwYACDBw+mf//+1eVTp05l3rx5vPTSSzUGS35+Ph9//HH1+9LSUtq3b1+9rH///pgZPXv2pEmTJqxfv568vJ1Gn6/Vtm3bmDNnDm+//XYdzk5EJD7UxLcH3J1LL72Uzp07c/XVV1eXz58/nwkTJvDss8+y//7717htjx49+OCDD/jwww/5/vvvmTFjBmeffTYA5557LgsWBDdM33//fb7//nvatGmzR3V78cUX6dSpE/n5+Xt5diIi8ZKxV1CpdAuP2uuvv86jjz5KYWEhxx9/PAC///3vufLKK9myZQt9+/YFgo4S9913H59++inDhw/nz3/+Mzk5OUycOJF+/fpRWVnJsGHDOPbYY4Gg+/iwYcPo0qULzZo1Y+rUqZjZDtsDDBo0iJdffpn169eTn5/PLbfcwqWXXgrAjBkz1LwnIlnFaro3stNKZqOAoUAhMN3dh4blg4H7E1ZtAuwHFLn722Y2BvgNsCVhnePcfc2ujldUVOTJExauXLmSzp0777auUjt9hiLxoW7m/2Jmb7t7UXJ5qldQnwJjgX4EAQSAu08DpiUcZChwI/BOwrYz3f3ne1FnEZFYWdkpwh94J02Kbl9ZKqWAcvc5AGZWBOzqJscQ4BFP5bJMRERkFyLrJGFmRwB9gEeSFp1lZl+a2XIz+2VUxxMRkewWZS++i4FX3f3DhLJZQGcgD7gMuMnMaryTb2YjzKzEzErKysoirJaIiGSiqANqamKBu69w90/dvdLd3wD+CJxf08buPtndi9y9aE+e/xERkewUSUCZ2QlAe+CJ3azqgCYmEhGR3Uqpk4SZ5YTrNgWamlkusM3dt4WrDAGedPfypO3OAV4BNgI9gCuBX0dS8zERj9g9ZtNuV4nrdBuLFy9m5MiRbN68mZycHO655x569uy5d5+DiEhMpHoFdQPwHVAM/Dx8fQNAGFYDSWreC10E/DdQTtB5YoK717ReRqiabmPlypW8+eabTJo0iRUrVtC3b1+WLVvGu+++S8eOHRk3btxO21ZNt/H888+zYsUKpk+fzooVKwB2mG5j+fLlXHvttTUef+jQocyfP3+n8uuuu46bb76ZxYsXc+utt3LddddFe+IiImmQajfzMcCYWpZtBlrVsiyrhjZo164d7dq1A3aebqNK7969eeKJnVs6E6fbAKqn2zjmmGP2aLqNtWvX7lRuZnz99dcAbNq0qXqMPxGRTKax+PZSnKbbuOuuuxg9ejSHHXYY1157bY1XcCIimUYBtRfqc7qN22+/nYEDB9a4fm3uvfde7rzzTj7++GPuvPPO6vH5REQymQJqD+1uuo1p06ZFNt1GqqZOnVpdlwsuuIC33nprb09PRCQ2FFB7IK7TbbRv356//e1vACxYsEAz6opIVsjY6TZS6RYetbhOt/HAAw9w1VVXsW3bNnJzc5k8eXKDfzYiIlFLabqNhqbpNuqHPkORuolyNPMFEY5mnq3TbaiJT0REYkkBJSIisaSAEhGRWFJAiYhILCmgREQklhRQIiISSxn7HFTh1MJI97d0yNLdrhPX6TaWLFnCyJEjqaiooKCggGnTpu0wBJOISCbSFdQeiOt0G8OHD2f8+PEsXbqU8847j9tvvz3aExcRSQMF1B5o164d3bp1A3aebiMnJ7gY7d27N6WlpTttmzjdRrNmzaqn2wD2aLqNgw8+eKfyVatW0adPHwD69u3Lk08+WfeTFRFJMwXUXorTdBtdunTh2WefBWD27Nk7DEorIpKpFFB7IW7TbTz00ENMmjSJ7t27U15eTrNmzfbirERE4iVjO0mky+6m23jppZcim24jLy8vpTp16tSJv/71r0BwNfbcc8/V5RRFRGIhpSsoMxtlZiVmtsXMHk4oLzAzN7OKhL8bE5abmU0wsw3h321W03+9M0Rcp9v44osvANi+fTtjx45l5MiRe3uKIiKxkeoV1KfAWKAfsF8Ny1u5+7YaykcA5wJdAQdeANYA9+1xTZOk0i08anGdbmP69OlMmhSMjNy/f38uueSSBv9sRESitkfTbZjZWCDf3YeG7wuAD4F9agooM3sDeNjdJ4fvLwUuc/feuzqOptuoH/oMRepG023Uj/qebuMjMys1sylmltg2dSywJOH9krCspgqOCJsRS8rKyiKqloiIZKq6BtR6oAdwBNAdaA5MS1h+IJA49e0m4MCa7kO5+2R3L3L3olQ7B4iISPaqUy8+d68AqtriPjezUcA6M2vh7l8DFUDimDstgAqP4zS+IiISK1E/B1UVPFVXSMsJOkhU6RqWiYiI7FKq3cxzzCwXaAo0NbPcsKyXmf3IzJqYWWvgT8DL7l7VrPcIcLWZHWpm7YFrgIfr4TxERCTLpHoFdQPwHVAM/Dx8fQNwFDAfKAeWAVuAQQnb3Q/MBZaGy58Ly0RERHYppXtQ7j4GGFPL4um72M6B68K/SEXZ3ROg83srd7tOOqfbqO3YAF9++SUXXngha9eupaCggFmzZnHQQQfV7QMREUkzjcW3B9I53UZtxwYYP348p5xyCh988AGnnHIK48ePr98PQkSkASig9kA6p9uo7dgAzzzzDEOGDAFgyJAhPP3009GeuIhIGiig9lI6p9tIPvbnn39Ou3btgCDIqsbmExHJZBrNfC/U53QbixYtYuDAgaxZs6bGUdFrO7aISLbRFdQe2t10G9OmTYtsuo1Uj922bVvWrVsHwLp162qdkVdEJJMooPZAOqfbqO3YAGeffTZTp04FgqA855xzIjtnEZF0ydgmvlS6hUctndNt1Hbs008/neLiYgYOHMiDDz7I4YcfzuzZsxv8sxERidoeTbfRUDTdRv3QZyhSN5puo37U93QbIiIikVJAiYhILCmgREQkljK2k4SISCoKpxZGtq9Zke1JUqErKBERiSUFlIiIxFLGNvFNGrkg0v3trpvmxo0befzxx7n88stT3ufQoUM588wzOf/88+taPRGRRkdXUCnauHEj99xzT7qrISLSaCigUlRcXMzq1as5/vjjGT16NKNHj6ZLly4UFhYyc+ZMIBiOaNSoURxzzDGcccYZO4wqfuutt9KjRw+6dOnCiBEjcHdWr15dPYUGwAcffED37t0b/NxEROJIAZWi8ePH06FDBxYvXkzv3r1ZvHgxS5Ys4cUXX2T06NGsW7eOp556ilWrVrF06VIeeOAB3njjjertR40axaJFi1i2bBnfffcd8+bNo0OHDrRs2ZLFixcDMGXKFIYOHZqeExQRiZmUAsrMRplZiZltMbOHE8p7m9kLZvalmZWZ2Wwza5ewfIyZbTWzioS/o+rhPBrUa6+9xqBBg2jatClt27atnsPplVdeqS5v3749J5/8r/taCxcupFevXhQWFrJgwQKWL18OwPDhw5kyZQqVlZXMnDmTn/3sZ+k6LRGRWEn1CupTYCzwUFL5QcBkoAA4AigHpiStM9PdD0z4W1OH+sbCrsYvrGmqjc2bN3P55ZfzxBNPsHTpUi677DI2b94MwIABA3j++eeZN28e3bt3p3Xr1vVWbxGRTJJSQLn7HHd/GtiQVP68u89296/d/VtgInBC9NVMv+bNm1NeXg5Anz59mDlzJpWVlZSVlfHKK6/Qs2dP+vTpw4wZM6isrGTdunUsXLgQoDqM2rRpQ0VFBU888UT1fnNzc+nXrx+//OUvueSSSxr+xEREYirqbuZ9gOVJZWeZ2ZfAOmCiu99b04ZmNgIYAXD44Yfv9kANPXpv69atOeGEE+jSpQunnXYaxx13HF27dsXMuO222/jBD37Aeeedx4IFCygsLKRjx46ceOKJALRq1YrLLruMwsJCCgoK6NGjxw77Hjx4MHPmzOHUU09t0HMSEYmzyALKzI4DbgISZ8ubRdAE+DnQC3jSzDa6+/Tk7d19crguRUVF8ZsDBHj88cd3eH/77bfv8N7MmDhxYo3bjh07lrFjx9a47LXXXmPYsGE0bdo0moqKiGSBSALKzH4IPA9c5e6vVpW7+4qE1d4wsz8C5wM7BVRjdd5557F69erqGXVFRCRQ54AysyOAF4Hfuvuju1ndgZ17ETRiTz31VLqrICISS6l2M88xs1ygKdDUzHLDskOBBcAkd7+vhu3OMbODLNATuBJ4Zm8rG8fZfzOFPjsRyTSpdjO/AfgOKAZ+Hr6+ARgOHAXcnPisU8J2FwH/TdD9/BFggrtP3ZuK5ubmsmHDBv2Hdi+4Oxs2bCA3NzfdVRERSVlKTXzuPgYYU8viW3ax3aA9r1LN8vPzKS0tpaysLKpdNiq5ubnk5+enuxoiIinLmNHM99lnH4488sh0V0NERBqIxuITEZFYUkCJiEgsKaBERCSWFFAiIhJLCigREYklBZSIiMSSAkpERGJJASUiIrGkgBIRkVhSQImISCwpoEREJJYUUCIiEksKKBERiSUFlIiIxJICSkREYkkBJSIisZRSQJnZKDMrMbMtZvZw0rJTzOw9M/vWzBaa2REJy8zMJpjZhvDvNjOziM9BRESyUKpXUJ8CY4GHEgvNrA0wB7gROBgoAWYmrDICOBfoChwHnAn8ok41FhGRRiGlgHL3Oe7+NLAhaVF/YLm7z3b3zcAYoKuZdQqXDwHucPdSd/8EuAMYGkXFRUQku9X1HtSxwJKqN+7+DbA6LN9pefj6WGpgZiPCZsSSsrKyOlZLREQyXV0D6kBgU1LZJqB5Lcs3AQfWdB/K3Se7e5G7F+Xl5dWxWiIikunqGlAVQIukshZAeS3LWwAV7u51PK6IiGS5ugbUcoIOEACY2QFAh7B8p+Xh6+WIiIjsRqrdzHPMLBdoCjQ1s1wzywGeArqY2YBw+U3Au+7+XrjpI8DVZnaombUHrgEejvwsREQk66R6BXUD8B1QDPw8fH2Du5cBA4DfAV8BvYCLEra7H5gLLAWWAc+FZSIiIruUk8pK7j6GoAt5TcteBDrVssyB68I/ERGRlGmoIxERiSUFlIiIxJICSkREYkkBJSIisaSAEhGRWFJAiYhILCmgREQklhRQIiISSwooERGJpZRGkhARaVBjWka3ryMPj25f0qB0BSUiIrGkgBIRkVhSQImISCwpoEREJJYUUCIiEksKKBERiSUFlIiIxJICSkREYqnOAWVmFUl/lWZ2d7iswMw8afmNda+2iIhkuzqPJOHuB1a9NrMDgM+B2UmrtXL3bXU9loiINB5RN/GdD3wBvBrxfkVEpJGJOqCGAI+4uyeVf2RmpWY2xcza1LShmY0wsxIzKykrK4u4WiIikmkiCygzOxw4EZiaULwe6AEcAXQHmgPTatre3Se7e5G7F+Xl5UVVLRERyVBRjmZ+MfCau39YVeDuFUBJ+PZzMxsFrDOzFu7+dYTHFhGRLBNlE9/F7Hj1VJOqpj+L8LgiIpKFIgkoM/sJcChJvffMrJeZ/cjMmphZa+BPwMvuvimK44qISPaK6gpqCDDH3cuTyo8C5gPlwDJgCzAoomOKiEgWi+QelLv/opby6cD0KI4hIiKNi4Y6EhGRWFJAiYhILCmgREQklhRQIiISSwooERGJJQWUiIjEUpRDHYlII1ZQ/Fxk+1qbG9muJIPpCkpERGJJASUiIrGkgBIRkVhSQImISCwpoEREJJYUUCIiEksKKBERiSUFlIiIxJICSkREYkkBJSIisRRJQJnZy2a22cwqwr9VCctOMbP3zOxbM1toZkdEcUwREcluUV5BjXL3A8O/HwGYWRtgDnAjcDBQAsyM8JgiIpKl6ruJrz+w3N1nu/tmYAzQ1cw61fNxRUQkw0UZUOPMbL2ZvW5mJ4VlxwJLqlZw92+A1WG5iIhIraIKqOuBo4BDgcnAXDPrABwIbEpadxPQPHkHZjbCzErMrKSsrCyiaomISKaKJKDc/e/uXu7uW9x9KvA6cDpQAbRIWr0FUF7DPia7e5G7F+Xl5UVRLRERyWD1dQ/KAQOWA12rCs3sAKBDWC4iIlKrOgeUmbUys35mlmtmOWY2GOgD/AV4CuhiZgPMLBe4CXjX3d+r63FFRCS7RTHl+z7AWKATUAm8B5zr7qsAzGwAMBF4DPg7cFEExxQRkSxX54By9zKgxy6Wv0gQXiIiIinTUEciIhJLCigREYklBZSIiMSSAkpERGJJASUiIrGkgBIRkVhSQImISCwpoEREJJYUUCIiEksKKBERiSUFlIiIxJICSkREYkkBJSIisaSAEhGRWFJAiYhILCmgREQklhRQIiISSwooERGJpToHlJnta2YPmtlHZlZuZv8ws9PCZQVm5mZWkfB3Y92rLSIi2S4non18DJwI/BM4HZhlZoUJ67Ry920RHEtERBqJOl9Bufs37j7G3de6+3Z3nwd8CHSve/VERKSxivwelJm1BToCyxOKPzKzUjObYmZtatluhJmVmFlJWVlZ1NUSEZEME2lAmdk+wDRgqru/B6wHegBHEFxRNQ+X78TdJ7t7kbsX5eXlRVktERHJQFHcgwLAzJoAjwLfA6MA3L0CKAlX+dzMRgHrzKyFu38d1bFFRCT7RBJQZmbAg0Bb4HR331rLql61SRTHlfRZ2alzZPvq/N7KyPYlItkjqiuoe4HOwE/d/buqQjPrBWwEPgAOAv4EvOzumyI6roiIZKk6B5SZHQH8AtgCfBZcTEFYth34PXAI8DXwAjCorseU7DJp5ILI9nXFfSdHti8RSa86B5S7f8Sum+ym1/UYIiLS+ETWSULir3Bq4e5XStGsyPYkIlIzjcUnIiKxpIASEZFYUkCJiEgsKaBERCSW1Eki7sa0jG5fRx4e3b5EROqZAqoeFBQ/F9m+1uZGtisRkYyiJj4REYklBZSIiMSSAkpERGJJASUiIrGkgBIRkVhSQImISCwpoEREJJYUUCIiEksKKBERiSUFlIiIxJICSkREYqneA8rMDjazp8zsGzP7yMx+Vt/HFBGRzNcQg8VOAr4H2gLHA8+Z2RJ3X94AxxYRkQxVr1dQZnYAMAC40d0r3P014FngP+vzuCIikvnM3etv52Y/Bt5w9/0Syq4FTnT3s5LWHQGMCN/+CFhVbxWTXWkDrE93JUQamP7dp9cR7p6XXFjfTXwHApuSyjYBzZNXdPfJwOR6ro/shpmVuHtRuush0pD07z6e6ruTRAXQIqmsBVBez8cVEZEMV98B9T6QY2ZHJ5R1BdRBQkREdqleA8rdvwHmALea2QFmdgJwDvBofR5X6kTNrNIY6d99DNVrJwkInoMCHgL6AhuAYnd/vF4PKiIiGa/eA0pERGRvaKgjERGJJQWUiIjEkgJKRGQPmVnTdNehMVBAScrMrEn4v/pySqNlZjnuXmlmTczsajM7LN11ylYKKEmJmTV19+1m1hG4zMzapbtOIg0t/B5sC3+svQN0Bw5Ic7WylgJKdsvMLPzF2BX4L6AVsDW9tRJpGGaWZ2aHAITfAwPmAivdfbC7v2dmbcwsP1zf0lnfbKKAkt1ydw+fZ5sF3Ozu44ENZvaDpFFCRLKKmTUjCKPfm1nbsDgH+By4K1znAeARYKWZ/afr2Z3IKKCkVkm/BFsCnwIPmVlrYCHwADDPzAano34i9c3dvweuAHoDvzKzHwDbCFoR7jKzp4FCYBRwBzDEzPZNT22zT0NMWCgZKGxrrzSzQ4HmYTNGDrAEWAN8BNwK/F+gIG0VFalHZtbE3d8Of4TNApoC1wAXAv2AjQRTCm03s28IRsvZnq76ZhsFlOwk/FJWmtlxwPPAGDP7p7v/u5n1BT5393fDdQ8Dvk5nfUXqQ3jvdXv4fVhiZhcCM4E/EEzCOi9c71AzGwT8CjjZ3XV/NiIa6kiA6q6z2xLetwPeAO5y9z+GX9Lt4bKmwNHAGOBY4MeJ24pkuhq+D/u6+xYz6w48TvDDbby7f2ZmVxLMEn6Zuy9OT42zk66ghPCq6Ggzuw8gDKIjgcXu/sdwNQ/XbUIwp9eZwMFAt7DbbVN3r2z42otEK6kr+b3AfgSdgu4Nm/t+RhBS283sFmAi8Li7a0beiKmThEBwFbQwDKaqS+r9gd7hc08QtL0DHAIcBtwH9HP3rVUPLjZojUUilNghqOohXKAEyANWEYTUDDPr6O5vA4MIrpp+E2yicKoPauKTauH9pDMJfh0CPAz8jeDX4RfhOtOBFe7+2/C9qVutZDIzOwroATwV9trDzCYAnd397PD948BpwBfAWe7+vpkdD5S7++r01Dz7qYlPSLi/1B84F9iHoNniSWAo0N3M3iO40joWuLhqW4WTZIGBwJqqcArNA2YAmNkUoAvwb8DTwEIz+6nuN9U/BVQjVhVMVZ0fws4QTQm6z7q7321mHwInAR0JupcPSWjWU8cIyXjhg+dVzXz9CDoHvQ40MbMzgB8TdASqNLM3gW/RSCoNQgHVSFWFU9i80QvYF3jS3f9gZluB08Iv7IPu/npiU17VTeQ0Vl8kEma2T/iDywjuKY0GbgOedvdyM9sOlAKHmdlPCVoQTnX3r9JX68ZDAdUIJYTT0cBSgqa8k4EzzWxeeOXUBPjfQCszu8Pdv6naXh0iJBuEP7q2hq0Gc4CRwA8JmrXNzJ4E1hN0DHqQYMSI/1A4NRwFVCMUhlM+8B9AsbvfFY4zNhw4x8xWh819eQS9l75NZ31FopbUuWccQZP2OuAmMxsDXBKWPRqOInEQ8Jm7/zM9NW6c1IuvEQrHCptJ0CvpUnd/LCxvBdwPfOXuI8MyCweLVW89yTpmNg4oAn7r7q8klI8B/p2go8QMdy9PTw0bNz0H1UgkDfxaCUwAVgDDqgrdfSPBF7LAzPYPyxROks1aAF2BU6v+zQO4+xiC+Z7OQv+dTBtdQWWpmkLFzFoCRvD/+1dm9mPgUeADYAhQQRBQle4+qKHrLFKfEnueJnX6uYXgaul+YK67f5uwTRs9hJs+CqgslDAS+X4EDxu+Ew78OpFgRIjvgPnu/v/CkHoCaA1MI7ghPMTdv9WVk2SLhO9EE+BOoDlQDkwKH7odSzClxv3AnxM7BUn66NI1yyR8EVsA7wOFYWeHp4DHgGsJeiT92sx+6+7/IHhQ8V2go7tfEIbTfgonyRb+r5lw3wLaAa8BxwPjzewMd7+BYGijawma+zQrbgwooLJIQjg1B/4OzHP3qUBn4FN3n+zu/+Xu04HzgAvM7H+FY4tdA7Qws0cB3P27dJ2HSFTCK6YqfYAN7j7Q3R8CziB4xmkYgLsXA38B3taPs3hQQGWJpHB6Bdjm7r8MF28HmoUP5VZ1mFhCMG11XrhOCXA50NPMHmzY2otEz8wODB+paBL2XD0AOMaCiTdx9wrgFuCkcIRy3P0mdSWPDwVUlkho1isBviQIpGvDjhHLCMbXG23BvDYe9tjbQvCcU9WYeu8AFwFj03EOIlExs+uAM8JHJ+YCPwUWEHQIuqFqPXffAMwn+LEmMaNOElkifBq+DJjt7r8ws9MIpsSY7O6/M7MjgVcJxhgrBdoSPBnfTSNDSLYxszsI7jEdBbzj7gPCloNhwDnAOuBuoC9wPfBv7v5hmqortVBAZZFwrpr3E96fRfAl/P/uPtbM2gMjCJ6K3wz8xjXZoGSRpK7kSwk6RAwDng+HNWpJcC/qaoLxJ7cD/yfsLCQxo4DKQlUDYIavq0LqAWBc1cjlCesqnCSrVIVU+G+/J8FgyI8Bz4ZN21XrNQFyE597knjRWHxZqCqcwtdzzcyBPwL7m9kEd/86YbnCSTJS8nN6Ce+fM7M1YSehuWZ2O8FI5duAx83sAiAfuEvhFG+6gmokwi/lhcAF6kIrmS5pJIhO7v5ewrI+BBMLPubuV4ZlEwhmzd0MnAD8xN2XN3jFZY8ooBqBXfzSFMk4SeH0F+Bldx+XtE5v4K/Awwkh1Z9gxJRXEwNN4ksBJSIZIymcngQOdffeZvYboIW7X5+wbk/gZeB2d785LRWWOtFzUCKSEZLC6WHgJILnmyD4b1lHM/t1wiYlwDzgRjP7bQNWVSKigBKRjJAQTk8ThNPXwKVmlgvcAbwI9DKzX4XrbwcWEUzE+Vgaqix1pF58IpIxzOxlYH93LzCzk4CpBEMY3QZMIZhO5lQzm0dwBXUV0MXdP0lLhaVOdA9KRDKGmXVz93cS3lc95zeZIKRygJ8AFxME11h3X5KOukrdKaBEJOOYWTN3/z58nRhSd1aNxB+OO7kljdWUOlITn4hknKpwCl/PDadv+gOQa2Z3uvtXwPe1bS+ZQZ0kRCTjuftcoJhgINgmYZmahzKcmvhEJONVdUEP54CqSHd9JBoKKBHJGholJbsooEREJJZ0D0pERGJJASUiIrGkgBIRkVhSQImISCwpoEREJJYUUCIiEksKKBERiSUFlIiIxNL/AM2Y0kaJ1TAcAAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "content_stats = global_stats.loc[['Topics', 'Tutorials']]\n",
    "# plot barplot stats\n",
    "ax = content_stats.plot(kind='bar', legend=True, fontsize=12)\n",
    "ax.set_xticklabels(content_stats.index, rotation=45)\n",
    "plt.tight_layout()\n",
    "plt.savefig(\"../images/topic_stat_plot.png\", transparent=True, dpi=150)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Contributors"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_contributors(ref):\n",
    "    '''\n",
    "    Extract the list of contributor at a given git commit\n",
    "\n",
    "    :param ref: name of the commit/branch/tag\n",
    "\n",
    "    :return: set with contributor ids\n",
    "    '''\n",
    "    file_content = repo.get_contents(\"CONTRIBUTORS.yaml\", ref=ref).content\n",
    "    decoded_file_content = base64.b64decode(file_content)\n",
    "    yaml_content = yaml.load(decoded_file_content, Loader=yaml.FullLoader)\n",
    "    contributors = set(yaml_content.keys())\n",
    "    return contributors"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "contributors = {}\n",
    "contr_stats = pd.DataFrame(columns=['Number of contributors'], index=[])\n",
    "for d, git_com in gitcommit_dates.items():\n",
    "    if d == '22.06.17':\n",
    "        continue\n",
    "    contributors[d] = get_contributors(git_com)\n",
    "    contr_stats.loc[d] = {'Number of contributors': len(contributors[d])}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Number of contributors</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>22.06.18</th>\n",
       "      <td>77</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22.06.19</th>\n",
       "      <td>122</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22.06.20</th>\n",
       "      <td>150</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>today</th>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         Number of contributors\n",
       "22.06.18                     77\n",
       "22.06.19                    122\n",
       "22.06.20                    150\n",
       "today                       157"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "contr_stats"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Contributions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "ename": "RateLimitExceededException",
     "evalue": "403 {\"message\": \"API rate limit exceeded for user ID 1842467.\", \"documentation_url\": \"https://docs.github.com/rest/overview/resources-in-the-rest-api#rate-limiting\"}",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mRateLimitExceededException\u001b[0m                Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-17-53067a620fe2>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m     30\u001b[0m         \u001b[0mreviewers\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0madd\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mr\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0muser\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mlogin\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     31\u001b[0m         \u001b[0mreview_nb\u001b[0m \u001b[0;34m+=\u001b[0m \u001b[0;36m1\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 32\u001b[0;31m     \u001b[0;32mfor\u001b[0m \u001b[0mr\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mpr\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_issue_comments\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     33\u001b[0m         \u001b[0mreviewers\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0madd\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mr\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0muser\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mlogin\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     34\u001b[0m         \u001b[0mcomment_nb\u001b[0m \u001b[0;34m+=\u001b[0m \u001b[0;36m1\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/envs/gtn-community-paper-2020/lib/python3.6/site-packages/github/PaginatedList.py\u001b[0m in \u001b[0;36m__iter__\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m     57\u001b[0m             \u001b[0;32myield\u001b[0m \u001b[0melement\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     58\u001b[0m         \u001b[0;32mwhile\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_couldGrow\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 59\u001b[0;31m             \u001b[0mnewElements\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_grow\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     60\u001b[0m             \u001b[0;32mfor\u001b[0m \u001b[0melement\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mnewElements\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     61\u001b[0m                 \u001b[0;32myield\u001b[0m \u001b[0melement\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/envs/gtn-community-paper-2020/lib/python3.6/site-packages/github/PaginatedList.py\u001b[0m in \u001b[0;36m_grow\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m     69\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     70\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_grow\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 71\u001b[0;31m         \u001b[0mnewElements\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_fetchNextPage\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     72\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__elements\u001b[0m \u001b[0;34m+=\u001b[0m \u001b[0mnewElements\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     73\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mnewElements\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/envs/gtn-community-paper-2020/lib/python3.6/site-packages/github/PaginatedList.py\u001b[0m in \u001b[0;36m_fetchNextPage\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    202\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_fetchNextPage\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    203\u001b[0m         headers, data = self.__requester.requestJsonAndCheck(\n\u001b[0;32m--> 204\u001b[0;31m             \u001b[0;34m\"GET\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__nextUrl\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mparameters\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__nextParams\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mheaders\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__headers\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    205\u001b[0m         )\n\u001b[1;32m    206\u001b[0m         \u001b[0mdata\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdata\u001b[0m \u001b[0;32mif\u001b[0m \u001b[0mdata\u001b[0m \u001b[0;32melse\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/envs/gtn-community-paper-2020/lib/python3.6/site-packages/github/Requester.py\u001b[0m in \u001b[0;36mrequestJsonAndCheck\u001b[0;34m(self, verb, url, parameters, headers, input)\u001b[0m\n\u001b[1;32m    317\u001b[0m         return self.__check(\n\u001b[1;32m    318\u001b[0m             *self.requestJson(\n\u001b[0;32m--> 319\u001b[0;31m                 \u001b[0mverb\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0murl\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mparameters\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mheaders\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0minput\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__customConnection\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0murl\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    320\u001b[0m             )\n\u001b[1;32m    321\u001b[0m         )\n",
      "\u001b[0;32m~/miniconda3/envs/gtn-community-paper-2020/lib/python3.6/site-packages/github/Requester.py\u001b[0m in \u001b[0;36m__check\u001b[0;34m(self, status, responseHeaders, output)\u001b[0m\n\u001b[1;32m    340\u001b[0m         \u001b[0moutput\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__structuredFromJson\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0moutput\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    341\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mstatus\u001b[0m \u001b[0;34m>=\u001b[0m \u001b[0;36m400\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 342\u001b[0;31m             \u001b[0;32mraise\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__createException\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mstatus\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mresponseHeaders\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0moutput\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    343\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mresponseHeaders\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0moutput\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    344\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mRateLimitExceededException\u001b[0m: 403 {\"message\": \"API rate limit exceeded for user ID 1842467.\", \"documentation_url\": \"https://docs.github.com/rest/overview/resources-in-the-rest-api#rate-limiting\"}"
     ]
    }
   ],
   "source": [
    "pr_stats = pd.DataFrame(\n",
    "    0,\n",
    "    columns=['PR', 'Reviews and comments (all PR)', 'Reviews and comments (mean per PR)', 'Reviewers (all PR)', 'Reviewers (mean per PR)', 'Duration (all PR)', 'Duration (mean per PR)'],\n",
    "    index=['PR to add tutorial', 'PR to update tutorial', 'Other PR'])\n",
    "pr_stats.loc[:, 'Duration (all PR)'] = datetime.timedelta(0)\n",
    "pr_stats.loc[:, 'Duration (mean per PR)'] = datetime.timedelta(0)\n",
    "prs = {} # keys: PR number as keys, values: PR object\n",
    "tuto_prs = {} # Keys: tutorials, values: set of PRs on the tutorial\n",
    "#n = 1\n",
    "for pr in repo.get_pulls(state=\"all\"):\n",
    "    # keep only merged PR\n",
    "    if not pr.merged:\n",
    "        continue\n",
    "    # try to identify PR for new tutorials\n",
    "    template_pr = False\n",
    "    status = 'Other PR'\n",
    "    for l in pr.labels:\n",
    "        if l.name == 'new tutorial':\n",
    "            status = 'PR to add tutorial'\n",
    "    prs[pr.number] = pr\n",
    "    # to test\n",
    "    #if n > 100:\n",
    "    #    break\n",
    "    #n += 1\n",
    "    # extract stats\n",
    "    review_nb = 0\n",
    "    comment_nb = 0\n",
    "    reviewers = set()\n",
    "    for r in pr.get_reviews():\n",
    "        reviewers.add(r.user.login)\n",
    "        review_nb += 1\n",
    "    for r in pr.get_issue_comments():\n",
    "        reviewers.add(r.user.login)\n",
    "        comment_nb += 1\n",
    "    # extract changed files\n",
    "    update = False\n",
    "    for f in pr.get_files():\n",
    "        if 'tutorials' in f.filename:\n",
    "            # update PR status\n",
    "            if status == 'Other PR':\n",
    "                status = 'PR to update tutorial'\n",
    "            if status != 'PR to add tutorial' and f.status == 'added': \n",
    "                if '.md' in f.filename or 'slides.html' in f.filename:\n",
    "                    status = 'PR to add tutorial'\n",
    "            # add tutorial to list of changed tutorials\n",
    "            tuto_n_m = re.search(r'([a-zA-Z0-9\\-_]+\\/tutorials\\/[a-zA-Z0-9\\-_]+)', f.filename)\n",
    "            if tuto_n_m:\n",
    "                tuto_n = tuto_n_m.group(0)\n",
    "                tuto_prs.setdefault(tuto_n, set())\n",
    "                tuto_prs[tuto_n].add(pr.number)\n",
    "    # add stats\n",
    "    pr_stats.loc[status, 'PR'] += 1\n",
    "    pr_stats.loc[status, 'Reviews and comments (all PR)'] += review_nb + comment_nb\n",
    "    pr_stats.loc[status, 'Reviewers (all PR)'] += len(reviewers)\n",
    "    pr_stats.loc[status, 'Duration (all PR)'] += (pr.merged_at-pr.created_at)\n",
    "# compute mean values\n",
    "pr_stats.loc[:, 'Reviews and comments (mean per PR)'] = pr_stats.loc[:,'Reviews and comments (all PR)']/pr_stats.loc[:, 'PR']\n",
    "pr_stats.loc[:, 'Reviewers (mean per PR)'] = pr_stats.loc[:,'Reviewers (all PR)']/pr_stats.loc[:, 'PR']\n",
    "pr_stats.loc[:, 'Duration (mean per PR)'] = pr_stats.loc[:,'Duration (all PR)']/pr_stats.loc[:, 'PR']\n",
    "# make copy\n",
    "all_tuto_prs = copy.deepcopy(tuto_prs)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Pull Request statistics"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "pr_stats"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "tuto_prs = copy.deepcopy(all_tuto_prs)\n",
    "# merge some tutorials that have been moved / renamed\n",
    "to_merge = [\n",
    "    ['admin/tutorials/heterogeneous', 'admin/tutorials/heterogeneous-compute'],\n",
    "    ['admin/tutorials/monitoring-maintenance', 'admin/tutorials/monitoring'],\n",
    "    ['Admin-Corner/tutorials/Database_Schema', 'admin/tutorials/database-schema'],\n",
    "    ['Admin-Corner/tutorials/galaxy_docker', 'admin/tutorials/galaxy-docker'],\n",
    "    \n",
    "    ['Assembly/tutorials/debruijn-graph-assembly', 'assembly/tutorials/debruijn-graph-assembly'],\n",
    "    ['Assembly/tutorials/general-introduction', 'assembly/tutorials/general-introduction'],\n",
    "\n",
    "    ['chip-seq/tutorials/chip-seq', 'epigenetics/tutorials/estrogen-receptor-binding-site-identification'],\n",
    "    ['chip-seq/tutorials/ChIPseq', 'epigenetics/tutorials/estrogen-receptor-binding-site-identification'],\n",
    "    ['chip-seq/tutorials/estrogen-receptor-binding-site-identification', 'epigenetics/tutorials/estrogen-receptor-binding-site-identification'],\n",
    "    ['chip-seq/tutorials/formation_of_super-structures_on_xi', 'epigenetics/tutorials/formation_of_super-structures_on_xi'],\n",
    "    ['chip-seq/tutorials/tal1-binding-site-identification', 'epigenetics/tutorials/tal1-binding-site-identification'],\n",
    "    ['ChIP-Seq/tutorials/ChIPseq', 'epigenetics/tutorials/estrogen-receptor-binding-site-identification'],\n",
    "    ['ChIP-Seq/tutorials/estrogen_receptor_binding_site_identification', 'epigenetics/tutorials/estrogen-receptor-binding-site-identification'],\n",
    "    ['ChIP-Seq/tutorials/TAL1_binding_site_identification', 'epigenetics/tutorials/tal1-binding-site-identification'],\n",
    "    \n",
    "    ['contributing/tutorials/setup-galaxy-for-training', 'instructors/tutorials/setup-galaxy-for-training'],\n",
    "\n",
    "    ['Dev-Corner/tutorials/data_source_integration', 'dev/tutorials/data-source-integration'],\n",
    "    ['Dev-Corner/tutorials/tool_integration', 'dev/tutorials/tool-integration'],\n",
    "    ['Dev-Corner/tutorials/visualizations', 'dev/tutorials/visualization-generic'],\n",
    "    ['dev/tutorials/visualization', 'dev/tutorials/visualization-generic'],\n",
    "    ['dev/tutorials/interactive-environment', 'dev/tutorials/interactive-environments'],\n",
    "    \n",
    "    ['Exome-Seq/tutorials/Diploid-variant-calling', 'variant-analysis/tutorials/dip'],\n",
    "    ['Exome-Seq/tutorials/Exome-Seq', 'variant-analysis/tutorials/exome-seq'],\n",
    "    \n",
    "    ['Genome-Annotation/tutorials/general-introduction', 'genome-annotation/tutorials/genome-annotation'],\n",
    "    \n",
    "    ['introduction/tutorials/galaxy-intro-collections', 'galaxy-data-manipulation/tutorials/collections'],\n",
    "    ['introduction/tutorials/galaxy-intro-get-data', 'galaxy-data-manipulation/tutorials/get-data'],\n",
    "    ['introduction/tutorials/galaxy-intro-rules', 'galaxy-data-manipulation/tutorials/upload-rules'],\n",
    "    ['introduction/tutorials/galaxy-intro-history', 'galaxy-ui/tutorials/history'],\n",
    "    ['introduction/tutorials/galaxy-intro-history-to-workflow', 'galaxy-ui/tutorials/history-to-workflow'],\n",
    "    ['introduction/tutorials/processing-many-samples-at-once', 'galaxy-data-manipulation/tutorials/processing-many-samples-at-once'],\n",
    "    ['Introduction/tutorials/galaxy-intro-101', 'introduction/tutorials/galaxy-intro-101'],\n",
    "    ['Introduction/tutorials/galaxy-intro-collections', 'galaxy-data-manipulation/tutorials/collections'],\n",
    "    ['Introduction/tutorials/galaxy-intro-get-data', 'galaxy-data-manipulation/tutorials/get-data'],\n",
    "    ['Introduction/tutorials/galaxy-intro-peaks2genes', 'introduction/tutorials/galaxy-intro-peaks2genes'],\n",
    "    ['Introduction/tutorials/Galaxy101-1', 'introduction/tutorials/galaxy-intro-101'],\n",
    "    ['Introduction/tutorials/Galaxy101-2', 'introduction/tutorials/galaxy-intro-101'],\n",
    "    ['Introduction/tutorials/igv', 'introduction/tutorials/igv-introduction'],\n",
    "    ['Introduction/tutorials/IGV_Introduction', 'introduction/tutorials/igv-introduction'],\n",
    "    ['Introduction/tutorials/processing_many_samples', 'galaxy-data-manipulation/tutorials/processing-many-samples-at-once'],\n",
    "    ['Introduction/tutorials/Processing-many-samples-at-once', 'galaxy-data-manipulation/tutorials/processing-many-samples-at-once'],\n",
    "    ['Introduction/tutorials/workflows', 'galaxy-ui/tutorials/workflow-parameters'],\n",
    "    \n",
    "    ['Metagenomics/tutorials/mothur-miseq-sop', 'metagenomics/tutorials/mothur-miseq-sop'],\n",
    "    \n",
    "    ['MethylC-Seq/tutorials/methylation-seq', 'epigenetics/tutorials/methylation-seq'],\n",
    "    ['MethylC-Seq/tutorials/methylc_seq', 'epigenetics/tutorials/methylation-seq'],\n",
    "    ['MethylC-Seq/tutorials/Methylation-Seq', 'epigenetics/tutorials/methylation-seq'],\n",
    "    \n",
    "    ['NGS-QC/tutorials/dive_into_qc', 'sequence-analysis/tutorials/quality-control'],\n",
    "    \n",
    "    ['NGS-mapping/tutorials/dive_into_mapping', 'sequence-analysis/tutorials/mapping'],\n",
    "    \n",
    "    ['Proteomics/tutorials/WIP_proteinQuant_SIL', 'proteomics/tutorials/protein-quant-sil'],\n",
    "    ['Proteomics/tutorials/database-handling', 'proteomics/tutorials/database-handling'],\n",
    "    ['Proteomics/tutorials/labelfree_vs_labelled', 'proteomics/tutorials/labelfree-vs-labelled'],\n",
    "    ['Proteomics/tutorials/proteinID_SG_PS', 'proteomics/tutorials/protein-id-sg-ps'],\n",
    "    ['Proteomics/tutorials/proteinQuant_SIL', 'proteomics/tutorials/protein-quant-sil'],\n",
    "    \n",
    "    ['RAD-Seq/tutorials/de_novo_rad', 'ecology/tutorials/de-novo-rad-seq'],\n",
    "    ['RAD-Seq/tutorials/genetic_map_rad', 'ecology/tutorials/genetic-map-rad-seq'],\n",
    "    ['RAD-Seq/tutorials/ref_based_rad', 'ecology/tutorials/ref-based-rad-seq'],\n",
    "    \n",
    "    ['RNA-Seq/tutorials/de_novo', 'transcriptomics/tutorials/de-novo'],\n",
    "    ['RNA-Seq/tutorials/ref_based', 'transcriptomics/tutorials/ref-based'],\n",
    "    ['RNA-Seq/tutorials/srna-seq', 'transcriptomics/tutorials/srna'],\n",
    "    ['RNA-Seq/tutorials/Reference-based-RNA-seq-detailed', 'transcriptomics/tutorials/ref-based'],\n",
    "    ['RNA-Seq/tutorials/RNA-Seq-hands-on', 'transcriptomics/tutorials/ref-based'],\n",
    "    \n",
    "    ['sequence-analysis/tutorials/ref-based-rad-seq', 'ecology/tutorials/ref-based-rad-seq'],\n",
    "    ['sequence-analysis/tutorials/de-novo-rad-seq', 'ecology/tutorials/de-novo-rad-seq'],\n",
    "    ['sequence-analysis/tutorials/genetic-map-rad-seq', 'ecology/tutorials/genetic-map-rad-seq'],\n",
    "    ['sequence-analysis/tutorials/genome-annotation', 'genome-annotation/tutorials/genome-annotation'],\n",
    "    ['sequence-analysis/tutorials/annotation-with-prokka', 'genome-annotation/tutorials/annotation-with-prokka' ],\n",
    "    ['sequence-analysis/tutorials/srna', 'transcriptomics/tutorials/srna'],\n",
    "    \n",
    "    ['training/tutorials/create-new-tutorial-content', 'contributing/tutorials/create-new-tutorial-content'],\n",
    "    ['training/tutorials/create-new-tutorial-tours', 'contributing/tutorials/create-new-tutorial-tours'],\n",
    "    ['training/tutorials/create-new-tutorial-jekyll', 'contributing/tutorials/running-jekyll'],\n",
    "    \n",
    "    ['transcriptomics/tutorials/scrna_preprocessing', 'transcriptomics/tutorials/scrna-preprocessing'],\n",
    "    ['transcriptomics/tutorials/limma-voom', 'transcriptomics/tutorials/rna-seq-counts-to-genes',],\n",
    "    ['transcriptomics/tutorials/limma-voom_fastqs_to_counts', 'transcriptomics/tutorials/rna-seq-reads-to-counts'],\n",
    "    \n",
    "    ['Usegalaxy/tutorials/collections', 'galaxy-data-manipulation/tutorials/collections'],\n",
    "    ['Usegalaxy/tutorials/dip', 'variant-analysis/tutorials/dip'],\n",
    "    ['Usegalaxy/tutorials/dunovo', 'variant-analysis/tutorials/dunovo'],\n",
    "    ['Usegalaxy/tutorials/history', 'galaxy-ui/tutorials/history'],\n",
    "    ['Usegalaxy/tutorials/ngs', 'introduction/tutorials/galaxy-intro-ngs-data-managment'],\n",
    "    ['Usegalaxy/tutorials/non_dip', 'variant-analysis/tutorials/non-dip'],\n",
    "    ['Usegalaxy/tutorials/rb_rnaseq', 'transcriptomics/tutorials/rb-rnaseq'],\n",
    "    ['usegalaxy/tutorials/collections', 'galaxy-data-manipulation/tutorials/collections'],\n",
    "    ['usegalaxy/tutorials/dip', 'variant-analysis/tutorials/dip'],\n",
    "    ['usegalaxy/tutorials/rb-rnaseq', 'transcriptomics/tutorials/rb-rnaseq'],\n",
    "    ['usegalaxy/tutorials/ngs', 'introduction/tutorials/galaxy-intro-ngs-data-managment'],\n",
    "    ['usegalaxy/tutorials/dunovo', 'variant-analysis/tutorials/dunovo'],\n",
    "    ['usegalaxy/tutorials/history', 'galaxy-ui/tutorials/history'],\n",
    "    ['usegalaxy/tutorials/non-dip', 'variant-analysis/tutorials/non-dip'],\n",
    "\n",
    "    ['variant-analysis/tutorials/diploid-variant-calling', 'variant-analysis/tutorials/dip'],\n",
    "]\n",
    "for t in to_merge:\n",
    "    tuto_prs[t[1]].update(tuto_prs[t[0]])\n",
    "    del tuto_prs[t[0]]\n",
    "    \n",
    "    \n",
    "to_remove = [\n",
    "    'Introduction/tutorials/images',\n",
    "    'Usegalaxy/tutorials/cp_img',\n",
    "    'templates/tutorials/tutorial',\n",
    "    'templates/tutorials/tutorial1',\n",
    "    \n",
    "    'Assembly/tutorials/index',\n",
    "    'Assembly/tutorials/tutorial',\n",
    "    'Dev-Corner/tutorials/files',\n",
    "    \n",
    "    # removed tutorials\n",
    "    'admin/tutorials/deployment-platforms-options',\n",
    "    'admin/tutorials/dev-to-production',\n",
    "    'admin/tutorials/job-conf',\n",
    "    'admin/tutorials/webservers',\n",
    "    'admin/tutorials/upstream-auth',\n",
    "    'Admin-Corner/tutorials/dev_to_production',\n",
    "    'Admin-Corner/tutorials/mysql_to_postgresql',\n",
    "    'Admin-Corner/tutorials/setting_up_galaxy_instance',\n",
    "    'contributing/tutorials/create-new-tutorial-metadata',\n",
    "    'contributing/tutorials/good-practices',\n",
    "    'dev/tutorials/github-contribution',\n",
    "    'introduction/tutorials/galaxy-intro-vis',\n",
    "    'Introduction/tutorials/galaxy-introduction',\n",
    "    'Introduction/tutorials/galaxy-intro-vis',\n",
    "    'Introduction/tutorials/introduction_my_first_analysis',\n",
    "    'Introduction/tutorials/Introduction',\n",
    "    'Introduction/tutorials/Introduction_to_Galaxy_Uni',\n",
    "    'metabolomics/tutorials/lcms-data-import',\n",
    "    'training/tutorials/create-new-tutorial-docker', \n",
    "    'training/tutorials/create-new-tutorial-metadata', \n",
    "    'training/tutorials/good-practices', \n",
    "]\n",
    "for t in to_remove:\n",
    "    del tuto_prs[t]\n",
    "    \n",
    "list(tuto_prs.keys())\n",
    "len(tuto_prs.keys())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Statistics per tutorials"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "tuto_pr_stats = pd.DataFrame(\n",
    "    0,\n",
    "    columns=['Creation', 'PR', 'Mean duration between PRs'],\n",
    "    index=[])\n",
    "for t in tuto_prs:\n",
    "    tuto_pr_stats.loc[t, 'PR'] = len(tuto_prs[t])\n",
    "    #tuto_pr_stats.loc[t, 'Commits'] = repo.get_commits(path = t).totalCount\n",
    "    # extract mean duration by sorting PRs and compute time between merge and creation\n",
    "    ordered_prs = sorted(tuto_prs[t])\n",
    "    tuto_pr_stats.loc[t, 'Creation'] = prs[ordered_prs[0]].merged_at\n",
    "    prev_pr_merged_at = prs[ordered_prs[0]].merged_at\n",
    "    duration = datetime.timedelta(0)\n",
    "    pr_n = 0\n",
    "    for pr in ordered_prs[1:]:\n",
    "        d = prs[pr].created_at - prev_pr_merged_at\n",
    "        # keep only positive delta\n",
    "        if d < datetime.timedelta(0):\n",
    "            continue\n",
    "        pr_n += 1\n",
    "        duration += d\n",
    "        prev_pr_merged_at = prs[pr].merged_at\n",
    "    if pr_n > 0:\n",
    "        tuto_pr_stats.loc[t, 'Mean duration between PRs'] = duration/pr_n\n",
    "    else:\n",
    "        tuto_pr_stats.loc[t, 'Mean duration between PRs'] = datetime.timedelta(0)\n",
    "tuto_pr_stats.loc[:, 'PR'] = tuto_pr_stats.loc[:, 'PR'].astype(int)\n",
    "tuto_pr_stats = (tuto_pr_stats\n",
    "    .sort_values('PR',ascending=False))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Top 10 tutorials in terms of PR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "tuto_pr_stats.head(12)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Mean number PR and commits per tutorials"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "tuto_pr_stats.mean()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Mean duration between PRs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "str(tuto_pr_stats.loc[:,'Mean duration between PRs'].sum()/tuto_pr_stats.shape[0])"
   ]
  }
 ],
 "metadata": {
  "hide_input": false,
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.11"
  },
  "toc": {
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {
    "height": "calc(100% - 180px)",
    "left": "10px",
    "top": "150px",
    "width": "165px"
   },
   "toc_section_display": true,
   "toc_window_display": true
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
